datafusion/common/src/error.rs

```diff
@@ -160,11 +158,11 @@ impl Display for SchemaError {
                 } else {
                     write!(f, "'{}'", field.name)?;
                 }
-                if let Some(fields) = valid_fields {
+                if !valid_fields.is_empty() {
```

Comment:
I'm wondering if we should add some additional text in case if it's empty?
Like `No data source passed`, `No valid fields` and etc. 

If we should add it, pls add your suggestion of error message 🤔

Reply:
I think it would be good to have a special case for this, otherwise it looks like the user would get the message `valid fields are `, which would be confusing.

Reply:
Currently (in this PR) it will just return `Schema error: No field named 'name'` (i.e `alid fields are ` won't be printed as before)

Let me know if I should add additional text for this case.